dsc_lib/src/discovery/discovery_trait.rs

```diff
@@ -5,6 +5,8 @@ use crate::{dscresources::dscresource::DscResource, dscerror::DscError};
 use std::collections::BTreeMap;
 
 pub trait ResourceDiscovery {
```

Comment:
To be clear on intent of the original 2 functions (it would be good to add this as a comment on these functions):

`list_available_resources` - used only in 'List' operation - it just returns the list of found classes (i.e. no caching);
`discover_resources` - used in all other operations - it searches for and caches requested classes in [Discovery.resources](https://github.com/PowerShell/DSC/blob/9ee037ae9a4491d1c8a932a3224c3a281a313cc0/dsc_lib/src/discovery/mod.rs#L13) so that later on (at the invocation time) class definition can be retrieved by simple/fast [`self.resources.get`](https://github.com/PowerShell/DSC/blob/9ee037ae9a4491d1c8a932a3224c3a281a313cc0/dsc_lib/src/discovery/mod.rs#L58).

Reply:
I'll add a comment.  These two functions still exist, but they both now use helpers to remove the duplicated code